templates/tapestry.js

```diff
@@ -901,6 +837,78 @@ function tapestryTool(config){
             rebuildNodeContents();
         }
     }
+
+    function initializeSelection() {
+        new DragSelect({
+            selectables: document.querySelectorAll(".node"),
+            onDragStart: () => {
+                if (!isMultiSelect) {
+                    selection.clear()
+                }
+            },
+            onElementSelect: node => {
+                const id = node.id.split("node-")[1]
+                selection.add(tapestry.dataset.nodes[findNodeIndex(id)])
+            },
+            onElementUnselect: node => {
+                const id = node.id.split("node-")[1]
+                selection.delete(tapestry.dataset.nodes[findNodeIndex(id)])
+            }
+        });
+    }
+
+    const selection = createSelection();
+    let isMultiSelect = false;
```

Comment:
Variables are all declared at the beginning of this file.

Reply:
Why should this be the case? For consistency?

I left it here because it's only being used in one place.